File: app/src/main/java/com/tommihirvonen/exifnotes/utilities/SquareImageView.java
package com.tommihirvonen.exifnotes.utilities;

import android.content.Context;
import androidx.annotation.Nullable;
import android.util.AttributeSet;

/**
 * Custom ImageView class, that forces the view to be square.
 */
public class SquareImageView extends androidx.appcompat.widget.AppCompatImageView {

    public SquareImageView(final Context context, @Nullable final AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(final int widthMeasureSpec, final int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        // Set the view height to be the same as the measured width.
        final int width = getMeasuredWidth();
        //noinspection SuspiciousNameCombination
        setMeasuredDimension(width, width);
    }

}

File: app/src/main/java/com/tommihirvonen/exifnotes/utilities/SquareImageView.kt
package com.tommihirvonen.exifnotes.utilities

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

/**
 * Custom ImageView class, that forces the view to be square.
 */
class SquareImageView(context: Context, attrs: AttributeSet?) : AppCompatImageView(context, attrs) {
    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        // Set the view height to be the same as the measured width.
        val width = measuredWidth
        setMeasuredDimension(width, width)
    }
}